src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -0,0 +1,237 @@
+    * getRecord method can accept into its 'fields' parameter.
+    */
+    buildFieldDescribesForWiredMethod(fields, objectApiName) {
+        return Object.keys(fields).map((fieldApiName) => {
+            return {
+                fieldApiName: fieldApiName,
+                objectApiName: objectApiName
+            }
+        });
+    }
+
+    /*******************************************************************************
+    * @description Contrcut any needed field info from the Sobject Info
+    */ 
+    buildDisplayedFields(fieldInfos) {
+        this.fields.name = this.extractFieldInfo(fieldInfos[FIELD_NAME.fieldApiName]);
+        this.fields.account = this.extractFieldInfo(fieldInfos[FIELD_ACCOUNT.fieldApiName]);
+        this.fields.contact = this.extractFieldInfo(fieldInfos[FIELD_CONTACT.fieldApiName]);
+        this.fields.recurringType = this.extractFieldInfo(fieldInfos[FIELD_RECURRING_TYPE.fieldApiName]);
+        this.fields.dateEstablished = this.extractFieldInfo(fieldInfos[FIELD_DATE_ESTABLISHED.fieldApiName]);
+        this.fields.status = this.extractFieldInfo(fieldInfos[FIELD_STATUS.fieldApiName]);
+        this.fields.amount = this.extractFieldInfo(fieldInfos[FIELD_AMOUNT.fieldApiName]);
+        this.fields.paymentMethod = this.extractFieldInfo(fieldInfos[FIELD_PAYMENT_METHOD.fieldApiName]);
+        this.fields.period = this.extractFieldInfo(fieldInfos[FIELD_INSTALLMENT_PERIOD.fieldApiName]);
+        this.fields.installmentFrequency = this.extractFieldInfo(fieldInfos[FIELD_INSTALLMENT_FREQUENCY.fieldApiName]);
+        this.fields.dayOfMonth = this.extractFieldInfo(fieldInfos[FIELD_DAY_OF_MONTH.fieldApiName]);
+        this.fields.startDate = this.extractFieldInfo(fieldInfos[FIELD_START_DATE.fieldApiName]);
+        this.fields.campaign = this.extractFieldInfo(fieldInfos[FIELD_CAMPAIGN.fieldApiName]);
+        this.fields.currency = {label: currencyFieldLabel, apiName: 'CurrencyIsoCode'};
+        this.fields.installment = this.extractFieldInfo(fieldInfos[FIELD_INSTALLMENT.fieldApiName]);
+    }
+
+    /*******************************************************************************
+    * @description Method converts field describe info into a object that is easily accessiable from the front end
+    */
+    extractFieldInfo(field) {
+        return {
+            apiName : field.apiName,
+            label : field.label,
+            inlineHelpText : field.inlineHelpText,
+            dataType : field.dataType
+        };
+    }
+
+    /*******************************************************************************
+    * @description Retrieve Recurring Donation record value
+    */ 
+    @wire(getRecord, { recordId: '$recordId', fields: '$fieldInfos' })
+    wiredRecurringDonationRecord(response) {
+        if (response.data) {
+            this.record = response.data;
+        }
+    }
+
+    async connectedCallback() {
+        getSetting({ recordId: this.recordId, parentId: this.parentId}).then(response => {
```

Comment:
Let's ensure `catch(error)` is used and follow formatting for method call/the/catch.
For example:
```
        getSetting({ recordId: this.recordId, parentId: this.parentId})
            .then((data) => {
                //code to execute based on data
            })
            .catch((error) => {
                this.handleError(error);
            });
```